src/Build/Evaluation/Expander.cs

```diff
@@ -4542,12 +4546,12 @@ private static void ConstructFunction(IElementLocation elementLocation, string e
                 }
 
                 // either there are no functions left or what we have is another function or an indexer
-                if (String.IsNullOrEmpty(remainder) || remainder[0] == '.' || remainder[0] == '[')
+                if (remainder.IsEmpty || remainder[0] == '.' || remainder[0] == '[')
                 {
-                    functionBuilder.Name = functionName;
+                    functionBuilder.Name = functionName.ToString();
                     functionBuilder.Arguments = functionArguments;
                     functionBuilder.BindingFlags = defaultBindingFlags;
-                    functionBuilder.Remainder = remainder;
+                    functionBuilder.Remainder = remainder.ToString();
```

Comment:
If this still shows as a hot spot for allocations, we'll probably have to tackle `FunctionBuilder`'s need to make these into real strings. But what you've done should help quite a bit and is a great start, so I think we should take this as-is.